tests/merge-context-workflow.test.js

```diff
@@ -0,0 +1,168 @@
+  })
+
+  afterEach(() => {
+    jest.resetModules()
+  })
+
+  afterAll(() => {
+    jest.clearAllMocks()
+  })
+
+  async function prepareWordItem (word, lang = Constants.LANG_LATIN) {
+    let langCode = LMF.getLanguageCodeFromId(lang)
+    /*
+    let adapterTuftsRes = await ClientAdapters.morphology.tufts({
+    method: 'getHomonym',
+    params: {
+        languageID: lang,
+        word: word
+    }
+    })
+
+    let testHomonym = adapterTuftsRes.result
+    */
+    let context = []
+    let tqselector = TextQuoteSelector.readObject({
+    languageCode: langCode,
+    targetWord: word,
+    target: {
+        source: 'foosource',
+        selector: {
+        exact: word,
+        prefix: 'fooprefix',
+        suffix: 'foosuffix'
+        }
+    }
+    })
+    context.push(tqselector)
+
+    return new WordItem({
+    targetWord: word,
+    languageCode: langCode,
+    // homonym: testHomonym,
+    context: context
+    })
+  }
+
+  it.skip('1 MergeContextWorkflow - createAbsentRemoteItems, createAbsentLocalItems', async () => {
+    let dbDriverRemote = new WordItemRemoteDbDriver('alpheiosMockUser')
+    let remoteAdapter = new RemoteDBAdapter(dbDriverRemote)
+
+    let dbDriverLocal = new WordItemIndexedDbDriver('alpheiosMockUser')
+    let localAdapter = new IndexedDBAdapter(dbDriverLocal)
+
+    let udm = new UserDataManager('alpheiosMockUser')
+
+    await udm.deleteMany({ dataType: 'WordItem', params: { languageCode: 'lat' }})
+
+    let testWord1 = await prepareWordItem('caeli')
+    testWord1.context[0].source = 'foosource1'
+    testWord1.context[0].prefix = 'fooprefix1'
+    testWord1.context[0].suffix = 'foosuffix1'
+
+    let testWord2 = await prepareWordItem('caeli')
+    testWord2.context[0].source = 'foosource2'
+    testWord2.context[0].prefix = 'fooprefix2'
+    testWord2.context[0].suffix = 'foosuffix2'
+
+    await udm.create({ dataObj: testWord1 }, { onlyLocal: true })
+    
+    await udm.create({ dataObj: testWord2 }, { onlyRemote: true })
+
+    let remoteItems = await udm.query({ dataType: 'WordItem', params: { languageCode: 'lat' }}, 'remote')
+    expect(remoteItems[0].context.length).toEqual(1)
+    expect(remoteItems[0].context[0].target.source).toEqual('foosource2')
+
+    let localItems = await udm.query({ dataType: 'WordItem', params: { languageCode: 'lat' }}, 'local')
+    expect(localItems[0].context.length).toEqual(1)
+    expect(localItems[0].context[0].source).toEqual('foosource1')
+    
+    await udm.createAbsentRemoteItems(localAdapter, remoteAdapter, localItems, remoteItems)
+    
+    remoteItems = await udm.query({ dataType: 'WordItem', params: { languageCode: 'lat' }}, 'remote')
+    expect(remoteItems[0].context.length).toEqual(2)
+    expect(remoteItems[0].context[0].target.source).toEqual('foosource2')
+    expect(remoteItems[0].context[0].target.selector.prefix).toEqual('fooprefix2')
+    expect(remoteItems[0].context[0].target.selector.suffix).toEqual('foosuffix2')
+
```

Comment:
should we test remoteItems[0].context[1] too?

Reply:
Yes, I should add a check for the second context! Fixed